### PR TITLE
Feature: Facebook Authentication with Chrome Custom Tabs

### PR DESCRIPTION
**Pull request by akueisara**
### Description

- [x] Create the `virtusize-core` module for multiple modules (virtusize and virtusize-auth) to reuse the common code
- [x] Demonstrate the step to enable Facebook authentication in clients' native mobile apps
- [x] Integrate `virtusize-auth.jar` for authentication code -> this jar will get deleted in PR #88 

Note: 
1. The general idea of the authentication flow is described here: https://www.notion.so/virtusize/Authentication-ee2385570ba14eb7969737c3c5f5cbc0#310dda9636a64347820a834b0120d4d4
2. A lot of unit tests are not passed in this PR. This will be fixed in PR #88 

## Demo

Nov-08-2021 18-44-11

### CODE REVIEW

File: virtusize/src/main/java/com/virtusize/android/ui/VirtusizeWebViewFragment.kt [hunk 1]
```diff
@@ -178,6 +179,11 @@ class VirtusizeWebViewFragment : DialogFragment() {
         binding.webView.loadUrl(virtusizeWebAppUrl)
     }
 
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        super.onActivityResult(requestCode, resultCode, data)
+        VirtusizeAuth.handleVirtusizeAuthResult(binding.webView, requestCode, resultCode, data)
```
Comment (akueisara) [inline virtusize/src/main/java/com/virtusize/android/ui/VirtusizeWebViewFragment.kt:184]: Please refer to https://github.com/virtusize/virtusize-auth-android/tree/feature/facebook-auth

File: virtusize/src/main/java/com/virtusize/android/ui/VirtusizeWebViewFragment.kt [hunk 2]
```diff
@@ -121,7 +122,7 @@ class VirtusizeWebViewFragment : DialogFragment() {
                                     return true
                                 }
                             }
-                            return false
+                            return VirtusizeAuth.isSNSAuthUrl(this@VirtusizeWebViewFragment, url)
```
Comment (akueisara) [inline virtusize/src/main/java/com/virtusize/android/ui/VirtusizeWebViewFragment.kt:125]: Please refer to https://github.com/virtusize/virtusize-auth-android/tree/feature/facebook-auth

File: sampleAppKotlin/src/main/AndroidManifest.xml [hunk 3]
```diff
@@ -21,6 +21,21 @@
             </intent-filter>
         </activity>
         <activity android:name=".ProductActivity" />
+        <activity
```
Comment (akueisara) [inline sampleAppKotlin/src/main/AndroidManifest.xml:24]: Client has to include this into their AndroidMainfest.xml to enable deep linking from Custom Custom Tabs back into the app

File: sampleAppKotlin/src/main/AndroidManifest.xml [hunk 4]
```diff
@@ -21,6 +21,21 @@
...
+                <action android:name="android.intent.action.VIEW" />
+
+                <category android:name="android.intent.category.DEFAULT" />
+                <category android:name="android.intent.category.BROWSABLE" />
+
+                <data
+                    android:host="sns-auth"
+                    android:scheme="com.virtusize.sampleappkotlin.virtusize" />
+            </intent-filter>
+        </activity>
```
Comment (akueisara) [inline sampleAppKotlin/src/main/AndroidManifest.xml:38]: Client has to include this into their AndroidMainfest.xml to enable deep linking from Custom Custom Tabs back into the app

File: sampleappjava/src/main/AndroidManifest.xml [hunk 5]
```diff
@@ -17,6 +17,20 @@
...
+            <intent-filter>
+                <action android:name="android.intent.action.VIEW" />
+                <category android:name="android.intent.category.DEFAULT" />
+                <category android:name="android.intent.category.BROWSABLE" />
+
+                <data
+                    android:host="sns-auth"
+                    android:scheme="com.virtusize.sampleappjava.virtusize" />
+            </intent-filter>
+        </activity>
```
Comment (akueisara) [inline sampleappjava/src/main/AndroidManifest.xml:33]: Clients have to include this into their AndroidMainfest.xml to handle deep links from the SNS (Facebook) auth page loaded in a Custom Custom Tab back into the app

File: sampleAppKotlin/src/main/AndroidManifest.xml [hunk 6]
```diff
@@ -21,6 +21,31 @@
...
+
+                <!--
+                    The URL host has to be sns-auth and the URL scheme must begin with your app's package ID (com.company.app) and end with .virtusize
+                    Note: The scheme you define must use all lowercase letters.
+                -->
+                <data
+                    android:host="sns-auth"
+                    android:scheme="com.virtusize.sampleappkotlin.virtusize" />
+            </intent-filter>
+        </activity>
```
Comment (akueisara) [inline sampleAppKotlin/src/main/AndroidManifest.xml:47]: Clients have to include this into their AndroidMainfest.xml to handle deep links from the SNS (Facebook) auth page loaded in a Custom Custom Tab back into the app

File: sampleappjava/src/main/AndroidManifest.xml (hunk 5 above)
Comment (zr0z) [reply sampleappjava/src/main/AndroidManifest.xml:33]: @akueisara can you mention those in the README?

**Review by zr0z — APPROVED** [review]
LGTM!

File: sampleappjava/src/main/AndroidManifest.xml (hunk 5 above)
Comment (akueisara) [reply sampleappjava/src/main/AndroidManifest.xml:33]: @zr0z Yes, I'm working on writing the README for the Android & iOS SDKs

**Pull request merged**